[2025-04-26T03:14:47.591+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-26T03:14:45.498085+00:00 [queued]>
[2025-04-26T03:14:47.610+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-26T03:14:45.498085+00:00 [queued]>
[2025-04-26T03:14:47.611+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-04-26T03:14:47.643+0000] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): submit_traffic_processor_spark_job> on 2025-04-26 03:14:45.498085+00:00
[2025-04-26T03:14:47.655+0000] {standard_task_runner.py:60} INFO - Started process 441 to run task
[2025-04-26T03:14:47.663+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'traffic_data_pipeline', 'submit_traffic_processor_spark_job', 'manual__2025-04-26T03:14:45.498085+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/traffic_pipeline_dag.py', '--cfg-path', '/tmp/tmpwhml2tpx']
[2025-04-26T03:14:47.674+0000] {standard_task_runner.py:88} INFO - Job 11: Subtask submit_traffic_processor_spark_job
[2025-04-26T03:14:47.762+0000] {task_command.py:423} INFO - Running <TaskInstance: traffic_data_pipeline.submit_traffic_processor_spark_job manual__2025-04-26T03:14:45.498085+00:00 [running]> on host b1fff31bea55
[2025-04-26T03:14:48.040+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='traffic_data_pipeline' AIRFLOW_CTX_TASK_ID='submit_traffic_processor_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-04-26T03:14:45.498085+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-26T03:14:45.498085+00:00'
[2025-04-26T03:14:48.043+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-26T03:14:48.046+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n            export KAFKA_BROKERS_INTERNAL="kafk-1:9092,kafk-2:9092,kafk-3:9092" && \\\n            export KAFKA_TOPIC="raw_traffic_data" && \\\n            export SCHEMA_REGISTRY_URL="http://schema-registry:8081" && \\\n            export REDIS_HOST="redis" && \\\n            export CLICKHOUSE_HOST="clickhouse-server" && \\\n            export CLICKHOUSE_NATIVE_PORT="9000" && \\\n            export MINIO_ENDPOINT="http://minio:9000" && \\\n            export MINIO_ACCESS_KEY="minioadmin" && \\\n            export MINIO_SECRET_KEY="minioadmin" && \\\n            export MINIO_BUCKET="traffic-data" && \\\n            export CHECKPOINT_LOCATION="s3a://traffic-data/checkpoints_***" && \\\n            export PYSPARK_PYTHON=/usr/bin/python3 && \\\n            spark-submit \\\n            --master local[*] \\\n            --deploy-mode client \\\n            --conf spark.sql.streaming.schemaInference=true \\\n            --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 \\\n            --conf spark.hadoop.fs.s3a.access.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.secret.key=minioadmin \\\n            --conf spark.hadoop.fs.s3a.path.style.access=true \\\n            --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem \\\n            --conf spark.sql.avro.schemaRegistryUrl=http://schema-registry:8081 \\\n            /opt/spark/app/processor.py\n            ']
[2025-04-26T03:14:48.058+0000] {subprocess.py:86} INFO - Output:
[2025-04-26T03:14:48.077+0000] {subprocess.py:93} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-04-26T03:14:51.267+0000] {subprocess.py:93} INFO - --- RUNNING LATEST processor.py with IMPORTED haversine ---
[2025-04-26T03:14:51.371+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SparkContext: Running Spark version 3.3.4
[2025-04-26T03:14:51.435+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-04-26T03:14:51.534+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceUtils: ==============================================================
[2025-04-26T03:14:51.535+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-04-26T03:14:51.536+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceUtils: ==============================================================
[2025-04-26T03:14:51.537+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SparkContext: Submitted application: TrafficProcessor
[2025-04-26T03:14:51.558+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-04-26T03:14:51.567+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceProfile: Limiting resource is cpu
[2025-04-26T03:14:51.568+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-04-26T03:14:51.623+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SecurityManager: Changing view acls to: ***
[2025-04-26T03:14:51.624+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SecurityManager: Changing modify acls to: ***
[2025-04-26T03:14:51.625+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SecurityManager: Changing view acls groups to:
[2025-04-26T03:14:51.625+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SecurityManager: Changing modify acls groups to:
[2025-04-26T03:14:51.626+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2025-04-26T03:14:51.890+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO Utils: Successfully started service 'sparkDriver' on port 46323.
[2025-04-26T03:14:51.923+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SparkEnv: Registering MapOutputTracker
[2025-04-26T03:14:51.960+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SparkEnv: Registering BlockManagerMaster
[2025-04-26T03:14:51.983+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-04-26T03:14:51.984+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-04-26T03:14:51.991+0000] {subprocess.py:93} INFO - 25/04/26 03:14:51 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-04-26T03:14:52.020+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-7f3d2cd7-81e6-44bf-85ac-15e0d62f3cb6
[2025-04-26T03:14:52.039+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-04-26T03:14:52.054+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-04-26T03:14:52.232+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-04-26T03:14:52.339+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO Executor: Starting executor ID driver on host b1fff31bea55
[2025-04-26T03:14:52.345+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-04-26T03:14:52.366+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 42551.
[2025-04-26T03:14:52.367+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO NettyBlockTransferService: Server created on b1fff31bea55:42551
[2025-04-26T03:14:52.368+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-04-26T03:14:52.374+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, b1fff31bea55, 42551, None)
[2025-04-26T03:14:52.378+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO BlockManagerMasterEndpoint: Registering block manager b1fff31bea55:42551 with 434.4 MiB RAM, BlockManagerId(driver, b1fff31bea55, 42551, None)
[2025-04-26T03:14:52.380+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, b1fff31bea55, 42551, None)
[2025-04-26T03:14:52.381+0000] {subprocess.py:93} INFO - 25/04/26 03:14:52 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, b1fff31bea55, 42551, None)
[2025-04-26T03:14:52.826+0000] {subprocess.py:93} INFO - Spark Session created. Version: 3.3.4
[2025-04-26T03:14:52.827+0000] {subprocess.py:93} INFO - Broadcasting segments data...
[2025-04-26T03:14:53.011+0000] {subprocess.py:93} INFO - Segments data broadcasted.
[2025-04-26T03:14:53.013+0000] {subprocess.py:93} INFO - Reading from Kafka: kafk-1:9092,kafk-2:9092,kafk-3:9092 / Topic: raw_traffic_data
[2025-04-26T03:14:53.014+0000] {subprocess.py:93} INFO - Kafka options: {'kafka.bootstrap.servers': 'kafk-1:9092,kafk-2:9092,kafk-3:9092', 'subscribe': 'raw_traffic_data', 'startingOffsets': 'latest', 'failOnDataLoss': 'false'}
[2025-04-26T03:14:55.896+0000] {subprocess.py:93} INFO - Reading Avro schema from file: /opt/spark/schemas/raw_traffic_event.avsc
[2025-04-26T03:14:55.901+0000] {subprocess.py:93} INFO - Successfully read Avro schema string.
[2025-04-26T03:14:55.902+0000] {subprocess.py:93} INFO - Deserializing Kafka messages using schema string from file...
[2025-04-26T03:14:56.404+0000] {subprocess.py:93} INFO - Using checkpoint location: s3a://traffic-data/checkpoints_***/traffic_processing_pipeline
[2025-04-26T03:14:56.621+0000] {subprocess.py:93} INFO - 25/04/26 03:14:56 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-04-26T03:14:57.750+0000] {subprocess.py:93} INFO - 25/04/26 03:14:57 WARN ResolveWriteToStream: spark.sql.adaptive.enabled is not supported in streaming DataFrames/Datasets and will be disabled.
[2025-04-26T03:14:58.016+0000] {subprocess.py:93} INFO - Streaming query 'traffic_processing_pipeline' started.
[2025-04-26T03:15:01.248+0000] {subprocess.py:93} INFO - 25/04/26 03:15:01 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-26T03:15:01.249+0000] {subprocess.py:93} INFO - 25/04/26 03:15:01 WARN KafkaDataConsumer: KafkaDataConsumer is not running in UninterruptibleThread. It may hang when KafkaDataConsumer's methods are interrupted because of KAFKA-1894
[2025-04-26T03:15:02.410+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-26T03:15:02.411+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.412+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-26T03:15:02.413+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-26T03:15:02.414+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.415+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-26T03:15:02.416+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-26T03:15:02.416+0000] {subprocess.py:93} INFO - EOFError
[2025-04-26T03:15:02.417+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.418+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.419+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.420+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.421+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.422+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.423+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.423+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.424+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.424+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage3.processNext(Unknown Source)
[2025-04-26T03:15:02.425+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.426+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.427+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.427+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.hashAgg_doAggregateWithKeys_0$(Unknown Source)
[2025-04-26T03:15:02.428+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
[2025-04-26T03:15:02.429+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.430+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.431+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.432+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
[2025-04-26T03:15:02.432+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2025-04-26T03:15:02.433+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2025-04-26T03:15:02.434+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2025-04-26T03:15:02.435+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:136)
[2025-04-26T03:15:02.435+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
[2025-04-26T03:15:02.436+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
[2025-04-26T03:15:02.436+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
[2025-04-26T03:15:02.437+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-04-26T03:15:02.438+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-04-26T03:15:02.439+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2025-04-26T03:15:02.439+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.440+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.441+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.442+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.443+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.444+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.445+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.446+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.446+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.447+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.448+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.449+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.450+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.451+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.452+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.453+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.454+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.454+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.455+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.456+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.457+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.457+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.458+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.459+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.460+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.461+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.462+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.463+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.464+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.464+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.465+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.466+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.467+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.468+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.469+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.470+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.471+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.471+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.472+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.473+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.474+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.475+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.476+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.476+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.477+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.478+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.478+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.479+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR PythonUDFRunner: Python worker exited unexpectedly (crashed)
[2025-04-26T03:15:02.480+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.480+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 666, in main
[2025-04-26T03:15:02.481+0000] {subprocess.py:93} INFO -     eval_type = read_int(infile)
[2025-04-26T03:15:02.482+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.482+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 595, in read_int
[2025-04-26T03:15:02.483+0000] {subprocess.py:93} INFO -     raise EOFError
[2025-04-26T03:15:02.484+0000] {subprocess.py:93} INFO - EOFError
[2025-04-26T03:15:02.484+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.485+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.486+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.486+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.487+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.488+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.489+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.489+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.490+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.491+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage3.processNext(Unknown Source)
[2025-04-26T03:15:02.491+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.492+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.493+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.493+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.hashAgg_doAggregateWithKeys_0$(Unknown Source)
[2025-04-26T03:15:02.494+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage4.processNext(Unknown Source)
[2025-04-26T03:15:02.495+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.495+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.496+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.496+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
[2025-04-26T03:15:02.497+0000] {subprocess.py:93} INFO - 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
[2025-04-26T03:15:02.498+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
[2025-04-26T03:15:02.498+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
[2025-04-26T03:15:02.499+0000] {subprocess.py:93} INFO - 	at org.apache.spark.scheduler.Task.run(Task.scala:136)
[2025-04-26T03:15:02.499+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)
[2025-04-26T03:15:02.500+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)
[2025-04-26T03:15:02.501+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)
[2025-04-26T03:15:02.501+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-04-26T03:15:02.502+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-04-26T03:15:02.503+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:840)
[2025-04-26T03:15:02.504+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.504+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.505+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.506+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.507+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.508+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.509+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.510+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.510+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.511+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.512+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.512+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.513+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.514+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.514+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.515+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.515+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.516+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.516+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.517+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.518+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.518+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.519+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.519+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.520+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.521+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.522+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.523+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.524+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.524+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.525+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.526+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.527+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.527+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.528+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.529+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.530+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.530+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.531+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.532+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.533+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.534+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.535+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.536+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.536+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.537+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.538+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.539+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-26T03:15:02.540+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.541+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.541+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.542+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.543+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.543+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.544+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.545+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.546+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.546+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.547+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.548+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.549+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.549+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.550+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.551+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.551+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.552+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.553+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.554+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.555+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.556+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.557+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.558+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.558+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.559+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.560+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.561+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.562+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.563+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.564+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.564+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.565+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.566+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.566+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.567+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.568+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.569+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.570+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.570+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.571+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.572+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.573+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.574+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.574+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.575+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.576+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.577+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR PythonUDFRunner: This may have been caused by a prior exception:
[2025-04-26T03:15:02.578+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.579+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.580+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.581+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.582+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.582+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.583+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.584+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.585+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.586+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.586+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.587+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.588+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.589+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.589+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.590+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.591+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.592+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.592+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.593+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.594+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.594+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.595+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.596+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.596+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.597+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.598+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.599+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.599+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.600+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.600+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.601+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.602+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.602+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.603+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.604+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.605+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.605+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.606+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.607+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.607+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.608+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.608+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.609+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.610+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.611+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.612+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.612+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR Executor: Exception in task 1.0 in stage 0.0 (TID 1)
[2025-04-26T03:15:02.613+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.613+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.614+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.615+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.615+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.616+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.617+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.617+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.618+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.618+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.619+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.620+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.620+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.621+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.621+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.622+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.623+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.623+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.624+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.625+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.625+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.626+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.627+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.627+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.628+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.629+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.630+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.630+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.631+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.632+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.632+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.633+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.633+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.634+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.635+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.635+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.636+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.637+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.637+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.638+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.639+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.639+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.640+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.640+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.641+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.642+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.642+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.643+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR Executor: Exception in task 0.0 in stage 0.0 (TID 0)
[2025-04-26T03:15:02.644+0000] {subprocess.py:93} INFO - org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.644+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.645+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.646+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.646+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.647+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.648+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.648+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.649+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.650+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.650+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.651+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.651+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.652+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.652+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.653+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.654+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.654+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.655+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.656+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.656+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.657+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.658+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.658+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.659+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.660+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.660+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.661+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.662+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.662+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.663+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.663+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.664+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.665+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.666+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.666+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.667+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.667+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.668+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.669+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.670+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.671+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.671+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.672+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.672+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.673+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.674+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.674+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 WARN TaskSetManager: Lost task 1.0 in stage 0.0 (TID 1) (b1fff31bea55 executor driver): org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-04-26T03:15:02.675+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.676+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.678+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.679+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.680+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.680+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.681+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.682+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.683+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.684+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.685+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.686+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.686+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.687+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.688+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.689+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.690+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.691+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.692+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.692+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.693+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:559)
[2025-04-26T03:15:02.694+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:86)
[2025-04-26T03:15:02.695+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$2.read(PythonUDFRunner.scala:68)
[2025-04-26T03:15:02.696+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:512)
[2025-04-26T03:15:02.697+0000] {subprocess.py:93} INFO - 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-04-26T03:15:02.697+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-04-26T03:15:02.698+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.699+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.700+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-04-26T03:15:02.700+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-04-26T03:15:02.701+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-04-26T03:15:02.701+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.702+0000] {subprocess.py:93} INFO - 	at org.apache.spark.ContextAwareIterator.hasNext(ContextAwareIterator.scala:39)
[2025-04-26T03:15:02.703+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.703+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.704+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.fill(Iterator.scala:1211)
[2025-04-26T03:15:02.704+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$GroupedIterator.hasNext(Iterator.scala:1217)
[2025-04-26T03:15:02.705+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-04-26T03:15:02.705+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach(Iterator.scala:943)
[2025-04-26T03:15:02.706+0000] {subprocess.py:93} INFO - 	at scala.collection.Iterator.foreach$(Iterator.scala:943)
[2025-04-26T03:15:02.706+0000] {subprocess.py:93} INFO - 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[2025-04-26T03:15:02.707+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.PythonRDD$.writeIteratorToStream(PythonRDD.scala:307)
[2025-04-26T03:15:02.708+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.python.PythonUDFRunner$$anon$1.writeIteratorToStream(PythonUDFRunner.scala:53)
[2025-04-26T03:15:02.708+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.$anonfun$run$1(PythonRunner.scala:438)
[2025-04-26T03:15:02.709+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2066)
[2025-04-26T03:15:02.710+0000] {subprocess.py:93} INFO - 	at org.apache.spark.api.python.BasePythonRunner$WriterThread.run(PythonRunner.scala:272)
[2025-04-26T03:15:02.710+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.711+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR TaskSetManager: Task 1 in stage 0.0 failed 1 times; aborting job
[2025-04-26T03:15:02.712+0000] {subprocess.py:93} INFO - 25/04/26 03:15:02 ERROR MicroBatchExecution: Query traffic_processing_pipeline [id = 40c16c78-8ded-4bdc-b67b-54870a3f0820, runId = 56e84b49-6895-4c21-bee4-32b980e08035] terminated with error
[2025-04-26T03:15:02.713+0000] {subprocess.py:93} INFO - py4j.Py4JException: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-26T03:15:02.713+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-26T03:15:02.713+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-26T03:15:02.714+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.714+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-26T03:15:02.715+0000] {subprocess.py:93} INFO -     raise e
[2025-04-26T03:15:02.715+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-26T03:15:02.716+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-26T03:15:02.717+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 264, in <lambda>
[2025-04-26T03:15:02.718+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-26T03:15:02.719+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.719+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 51, in write_to_redis
[2025-04-26T03:15:02.720+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-26T03:15:02.721+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-26T03:15:02.721+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-26T03:15:02.722+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-26T03:15:02.723+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.723+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-26T03:15:02.724+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-26T03:15:02.724+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.725+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 196, in deco
[2025-04-26T03:15:02.726+0000] {subprocess.py:93} INFO -     raise converted from None
[2025-04-26T03:15:02.726+0000] {subprocess.py:93} INFO - pyspark.sql.utils.PythonException:
[2025-04-26T03:15:02.727+0000] {subprocess.py:93} INFO -   An exception was thrown from the Python worker. Please see the stack trace below.
[2025-04-26T03:15:02.727+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-26T03:15:02.728+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.729+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.730+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.731+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.732+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.734+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.734+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.735+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.736+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.737+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.738+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.738+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.739+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.740+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.740+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.741+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.742+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.742+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.743+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.743+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.744+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.745+0000] {subprocess.py:93} INFO - 	at py4j.Protocol.getReturnValue(Protocol.java:476)
[2025-04-26T03:15:02.745+0000] {subprocess.py:93} INFO - 	at py4j.reflection.PythonProxyHandler.invoke(PythonProxyHandler.java:108)
[2025-04-26T03:15:02.746+0000] {subprocess.py:93} INFO - 	at jdk.proxy3/jdk.proxy3.$Proxy37.call(Unknown Source)
[2025-04-26T03:15:02.747+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1(ForeachBatchSink.scala:51)
[2025-04-26T03:15:02.748+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.PythonForeachBatchHelper$.$anonfun$callForeachBatch$1$adapted(ForeachBatchSink.scala:51)
[2025-04-26T03:15:02.748+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.sources.ForeachBatchSink.addBatch(ForeachBatchSink.scala:32)
[2025-04-26T03:15:02.749+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$17(MicroBatchExecution.scala:665)
[2025-04-26T03:15:02.750+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2025-04-26T03:15:02.751+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2025-04-26T03:15:02.752+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2025-04-26T03:15:02.753+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-26T03:15:02.754+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2025-04-26T03:15:02.755+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runBatch$16(MicroBatchExecution.scala:663)
[2025-04-26T03:15:02.756+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-26T03:15:02.756+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-26T03:15:02.757+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-26T03:15:02.758+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runBatch(MicroBatchExecution.scala:663)
[2025-04-26T03:15:02.758+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$2(MicroBatchExecution.scala:256)
[2025-04-26T03:15:02.759+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-26T03:15:02.760+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken(ProgressReporter.scala:375)
[2025-04-26T03:15:02.760+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProgressReporter.reportTimeTaken$(ProgressReporter.scala:373)
[2025-04-26T03:15:02.761+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:68)
[2025-04-26T03:15:02.762+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.$anonfun$runActivatedStream$1(MicroBatchExecution.scala:219)
[2025-04-26T03:15:02.763+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:67)
[2025-04-26T03:15:02.763+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:213)
[2025-04-26T03:15:02.764+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.$anonfun$runStream$1(StreamExecution.scala:307)
[2025-04-26T03:15:02.765+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-26T03:15:02.766+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2025-04-26T03:15:02.766+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:285)
[2025-04-26T03:15:02.767+0000] {subprocess.py:93} INFO - 	at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:208)
[2025-04-26T03:15:02.768+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-26T03:15:02.768+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 277, in <module>
[2025-04-26T03:15:02.769+0000] {subprocess.py:93} INFO -     main()
[2025-04-26T03:15:02.770+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 271, in main
[2025-04-26T03:15:02.770+0000] {subprocess.py:93} INFO -     streaming_query.awaitTermination()
[2025-04-26T03:15:02.771+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/streaming.py", line 107, in awaitTermination
[2025-04-26T03:15:02.773+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-26T03:15:02.773+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 196, in deco
[2025-04-26T03:15:02.774+0000] {subprocess.py:93} INFO - pyspark.sql.utils.StreamingQueryException: Query traffic_processing_pipeline [id = 40c16c78-8ded-4bdc-b67b-54870a3f0820, runId = 56e84b49-6895-4c21-bee4-32b980e08035] terminated with exception: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
[2025-04-26T03:15:02.775+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/clientserver.py", line 617, in _call_proxy
[2025-04-26T03:15:02.776+0000] {subprocess.py:93} INFO -     return_value = getattr(self.pool[obj_id], method)(*params)
[2025-04-26T03:15:02.777+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.778+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 276, in call
[2025-04-26T03:15:02.779+0000] {subprocess.py:93} INFO -     raise e
[2025-04-26T03:15:02.780+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 273, in call
[2025-04-26T03:15:02.780+0000] {subprocess.py:93} INFO -     self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
[2025-04-26T03:15:02.781+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 264, in <lambda>
[2025-04-26T03:15:02.782+0000] {subprocess.py:93} INFO -     write_to_redis(df, epoch_id),
[2025-04-26T03:15:02.783+0000] {subprocess.py:93} INFO -     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.783+0000] {subprocess.py:93} INFO -   File "/opt/spark/app/processor.py", line 51, in write_to_redis
[2025-04-26T03:15:02.784+0000] {subprocess.py:93} INFO -     if df.isEmpty(): return
[2025-04-26T03:15:02.785+0000] {subprocess.py:93} INFO -        ^^^^^^^^^^^^
[2025-04-26T03:15:02.786+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/dataframe.py", line 553, in isEmpty
[2025-04-26T03:15:02.787+0000] {subprocess.py:93} INFO -     return self._jdf.isEmpty()
[2025-04-26T03:15:02.788+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.789+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2025-04-26T03:15:02.790+0000] {subprocess.py:93} INFO -     return_value = get_return_value(
[2025-04-26T03:15:02.791+0000] {subprocess.py:93} INFO -                    ^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.792+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 196, in deco
[2025-04-26T03:15:02.793+0000] {subprocess.py:93} INFO -     raise converted from None
[2025-04-26T03:15:02.794+0000] {subprocess.py:93} INFO - pyspark.sql.utils.PythonException:
[2025-04-26T03:15:02.794+0000] {subprocess.py:93} INFO -   An exception was thrown from the Python worker. Please see the stack trace below.
[2025-04-26T03:15:02.795+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-26T03:15:02.796+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 670, in main
[2025-04-26T03:15:02.796+0000] {subprocess.py:93} INFO -     func, profiler, deserializer, serializer = read_udfs(pickleSer, infile, eval_type)
[2025-04-26T03:15:02.797+0000] {subprocess.py:93} INFO -                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.798+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 507, in read_udfs
[2025-04-26T03:15:02.799+0000] {subprocess.py:93} INFO -     udfs.append(read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index=i))
[2025-04-26T03:15:02.800+0000] {subprocess.py:93} INFO -                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.800+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 289, in read_single_udf
[2025-04-26T03:15:02.801+0000] {subprocess.py:93} INFO -     f, return_type = read_command(pickleSer, infile)
[2025-04-26T03:15:02.802+0000] {subprocess.py:93} INFO -                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.803+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/worker.py", line 85, in read_command
[2025-04-26T03:15:02.804+0000] {subprocess.py:93} INFO -     command = serializer._read_with_length(file)
[2025-04-26T03:15:02.804+0000] {subprocess.py:93} INFO -               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.805+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 173, in _read_with_length
[2025-04-26T03:15:02.806+0000] {subprocess.py:93} INFO -     return self.loads(obj)
[2025-04-26T03:15:02.807+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.808+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/serializers.py", line 471, in loads
[2025-04-26T03:15:02.808+0000] {subprocess.py:93} INFO -     return cloudpickle.loads(obj, encoding=encoding)
[2025-04-26T03:15:02.809+0000] {subprocess.py:93} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-04-26T03:15:02.810+0000] {subprocess.py:93} INFO - TypeError: code() argument 13 must be str, not int
[2025-04-26T03:15:02.811+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.811+0000] {subprocess.py:93} INFO - 
[2025-04-26T03:15:02.902+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-04-26T03:15:02.918+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-04-26T03:15:02.929+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=traffic_data_pipeline, task_id=submit_traffic_processor_spark_job, execution_date=20250426T031445, start_date=20250426T031447, end_date=20250426T031502
[2025-04-26T03:15:02.948+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 11 for task submit_traffic_processor_spark_job (Bash command failed. The command returned a non-zero exit code 1.; 441)
[2025-04-26T03:15:02.963+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-26T03:15:02.991+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
